src/plugins/intel_cpu/src/node.cpp

```diff
@@ -771,46 +771,61 @@ void Node::initDescriptor(const NodeConfig& config) {
     selectedPD->setConfig(updatedConfig);
 }
 
-void Node::prepareMemory(const std::vector<DnnlMemoryDescPtr>& intDescs) {
-    if (internalBlobs.size() != intDescs.size()) {
-        IE_THROW() << "Can't prepare memory for internal blob, internal blob and internal descs number do not match "
-                   << internalBlobs.size() << " vs " << intDescs.size();
+void Node::prepareMemory(const DnnlMemoryDescPtr& intDesc, size_t indx) {
+    size_t minSize = indx + 1;
+    if (internalBlobMemory.size() < minSize) {
+        internalBlobMemory.resize(minSize);
     }
 
-    internalBlobMemory.clear();
-    for (size_t i = 0; i < internalBlobs.size(); i++) {
-        const auto &internalBlob = internalBlobs[i];
+    if (minSize > internalBlobs.size()) {
+        IE_THROW() << "Can't prepare memory for internal blob, requested index: " << indx <<
+            " is out of bounds of the internalBlobs vector of size " << internalBlobs.size();
+    }
 
-        auto create = [&] () {
-            // TODO [DS]: internal blobs should be removed or rewritten using Memory object
-            auto newDesc = MemoryDescUtils::convertToDnnlBlockedMemoryDesc(internalBlob->getTensorDesc());
+    const auto &internalBlob = internalBlobs[indx];
 
-            Memory memory{ engine };
-            memory.Create(newDesc, internalBlob->buffer());
+    auto create = [&] () {
+        // TODO [DS]: internal blobs should be removed or rewritten using Memory object
+        auto newDesc = MemoryDescUtils::convertToDnnlBlockedMemoryDesc(internalBlob->getTensorDesc());
 
-            MemoryPtr _ptr = std::make_shared<Memory>(engine);
-            _ptr->Create(*intDescs[i]);
-            _ptr->SetData(memory);
+        Memory memory{ engine };
+        memory.Create(newDesc, internalBlob->buffer());
 
-            return _ptr;
-        };
+        MemoryPtr _ptr = std::make_shared<Memory>(engine);
+        _ptr->Create(intDesc);
+        node::Reorder::reorderData(memory, *_ptr, context->getParamsCache());
```

Comment:
concern here :

Will this check and subnormals processing be skipped? Since node::Reorder::reorderData replaces where "_ptr->SetData(memory);" once be.
https://github.com/openvinotoolkit/openvino/blob/c186ffdf0dfc927710c9369dc7e193819e2e1b44/src/plugins/intel_cpu/src/cpu_memory.cpp#L78

Reply:
Yep, good point, however the main intention of this PR is rather a bug fix. Thus I would like to preserve the previous behavior as much as possible. With no doubts,  we can optimize this method as you suggest, but I would prefer to do it in a separate PR targeting the next release.